src/pages/guides/tutorials/stats-addon.md

```diff
@@ -0,0 +1,1108 @@
+      })`,
+    ]);
+    // the for...in loop iterates over the keys of an object
+    for (const node in pageData.nodes) {
+      // for example,      ["Text", 1]
+      addRowToTable(table, [node, pageData.nodes[node]]);
+    }
+  });
+};
+
+export { rebuildTable };
+```
+
+#### Document Sandbox
+
+```js
+import addOnSandboxSdk from "add-on-sdk-document-sandbox";
+const { runtime } = addOnSandboxSdk.instance;
+
+import { editor } from "express";
+import { getNodeData } from "./utils";
+
+async function start() {
+  const panelUIProxy = await runtime.apiProxy("panel");
+
+  runtime.exposeApi({
+    async getDocumentData() {
+      const doc = editor.documentRoot;
+      let documentData = [];
+      for (const page of doc.pages) {
+        console.log("Page", page);
+        let pageData = {};
+        pageData.dimensions = {
+          width: page.width,
+          height: page.height,
+        };
+        pageData.nodes = getNodeData(page);
+        documentData.push(pageData);
+      }
+      console.log("documentData", documentData);
+      await panelUIProxy.createTable(documentData);
+    },
+  });
+
+  panelUIProxy.toggleStatus("document");
+}
+
+start();
+```
+
+#### Document Sandbox
+
+```js
+const increaseCount = (obj, type) => {
+  // If the type is already in the object, increase its count, otherwise set it to 1
+  if (obj.hasOwnProperty(type)) {
+    obj[type] += 1;
+  } else {
+    obj[type] = 1;
+  }
+};
+
+const getNodeData = (node, nodeData = {}) => {
+  if (node.type === "MediaContainer") {
+    return nodeData;
+  }
+
+  // Check if the current node has children and if they are not an empty array
+  if (node.allChildren && node.allChildren.length > 0) {
+    // Iterate over all children using for..of
+    for (const child of node.allChildren) {
+      // Increase the count for the current type
+      increaseCount(nodeData, child.type);
+
+      // Recursively call getNodeData for each child that has its own children
+      // ... unless it's a MediaContainer
+      if (
+        child.type !== "MediaContainer" &&
+        child.allChildren &&
+        child.allChildren.length > 0
+      ) {
+        getNodeData(child, nodeData);
+      }
+    }
+  }
+  return nodeData;
+};
+
+export { getNodeData };
+```
+
+[^0]: When referring to "metadata", I mean the dimensions and types of elements on each page. Custom metadata haven't been implemented in Adobe Express yet.
+
+[^1]: In my tests, the two SDKs are ready almost instantly and together—I couldn't tell which one is loaded firsts.
+
+[^2]: Designing data structures in the early stages of the development process is, in my humble opinion, a way to dodge a good deal of future headaches.
+
+[^3]:  A `"MediaContainer"` (say, an image) has as children both the `"ImageRectangle"` and the mask that crops it. At this stage of the Document API development, getting complex masks may throw an Error—hence, I've decided to use the `"MediaContainer"` type as a recursion's termination clause.
```

Comment:
Is this still true?

Reply:
It still seems to be the case. It depends, though. A heart-shaped crop is a mask of type "Path" and is processed fine, a pentagon is of type undefined and throws.